### PR TITLE
Sockets - Rosalyn

### PR DESCRIPTION
**Pull request by ranuseh**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model is what contained the data for our application. In this case, it is linked to the task database. It keeps track of the current state of the application like which tasks have been created.

Describe in your own words what the Controller is doing in Rails | The controller is what interfaces with the outside world. It receives events through HTTP requests, gathers data from the model and passes data to the view which gets rendered as HTML

Describe in your own words what the View is doing in Rails | Is the area where the display logic is contained. It turns data is HTML which is rendered by the browser.

Describe an edge-case controller test you wrote | Won't update the completion date of a task when you are going from a completed task to an incomplete. Essentially when you reverse the completeness of a task

What is the purpose of using strong params? (i.e. the params method in the controller) | Controllers are not permitted to pass parameters directly from the form into the ActiveRecord model instances. This is a security feature that ensures that a user cannot update important model attributes that they should not be updating. Strong params require that you provide explicit access to a set of attributes.

How are Rails migrations related to Rails models? | When you do a rails db:migration and you modify the scheme of the database it also updates the model representation of the schema.

Describe one area of Rails that are still unclear on | The database migrations are still a bit of a mystery to me. It currently seems like just something running in the background and I have no idea what that looks like.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Rails.application.routes.draw do
+  root 'tasks#index'
+
+  get '/tasks', to: 'tasks#index', as: 'tasks'
+  get '/tasks/new', to: 'tasks#new', as: 'new_task'
+  post '/tasks', to: 'tasks#create'
+
+  get '/tasks/:id', to: 'tasks#show', as: 'task'
+  get '/tasks/:id/edit', to: 'tasks#edit', as: 'edit_task'
+  patch '/tasks/:id', to: 'tasks#update'
```
Comment (dHelmgren) [inline config/routes.rb:10]: I would have liked to see you using `resources` to dry this up!

File: app/views/tasks/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<h2>Update Task Form</h2>
+
+<%= form_with model: @task, class: 'Update_task' do |f| %>
+<p><strong>Update Task Details Below</strong></p>
+
+  <div>
+    <%= f.label :name, "Name of Task:" %>
+    <%= f.text_field :name, placeholder: @task.name %>
+  </div>
+
```
Comment (dHelmgren) [inline app/views/tasks/edit.html.erb:10]: You've got almost exactly the same code here to render a form as you do in `new`. You should dry this up with a view partial.

**Comment by dHelmgren** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | yes
Answered comprehension questions | yes
Successfully handles: Index, Show | yes
Index & Show tests pass | yes
Successfully handles: New, Create | yes
New & Create tests pass | yes
Successfully handles: Edit, Update | yes
Tests for Edit & Update test valid & invalid task ids | yes
Successfully handles: Destroy, Task Complete | yes
Tests for Destroy & Task Complete include tests for valid and invalid task ids | yes
Routes follow RESTful conventions | yes
Uses named routes (like `_path`) | yes
**Overall** | Good job on Task List! It functions as expected. You hit the learning goals: using strong params, and following Rails conventions. Keep in mind where you can use View Partials for future assignments.
